tests/test_extractors.py

```diff
@@ -0,0 +1,64 @@
+from os import path
+from typing import cast
+
+import pytest
+import tempfile
+import torch_geometric.data as pyg_data
+import torch_geometric.datasets as pyg_dataset
+import qek.data.dataset as qek_dataset
+from qek.data.extractors import QutipExtractor, EmuMPSExtractor
+from qek.data.graphs import MoleculeGraphCompiler
+
+
+@pytest.mark.asyncio
+async def test_emulators() -> None:
```

Comment:
Why do you need asynchronicity here ?

Reply:
Yeah but that would only hold for the `QPUExtractor` ?

Reply:
Yes, but the idea is to have the same method `run()` for all extractors, to allow substituting one to the other transparently.

Reply:
I get that but it feels an unnecessary overhead to implement async calls for emulators IMHO.

Reply:
Do you have ideas for a better design? Any suggestions welcome :)

Reply:
If that's ok, I'll try and rework this in a followup.